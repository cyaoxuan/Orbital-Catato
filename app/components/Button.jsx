import { View } from "react-native";
import { Button } from "react-native-paper";

const PillButton = (props) => {
    return (
        <View style={{ flexDirection: "row" }}>
                <Button style = {{ margin: 10, width: props.width }}
                    disabled={props.disabled}
                    mode={props.mode}
                    onPress={props.onPress}>
                        {props.label}
                </Button>
        </View>
    )
}

export { PillButton };