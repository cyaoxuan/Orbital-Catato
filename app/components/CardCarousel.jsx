import { View, FlatList } from "react-native";
import { CatCard } from "../components/CatCard";

const boxWidth = 300;

const Carousel = (props) => {
    return (
        <FlatList style={{ height: 300 }}
                horizontal
                snapToAlignment="center"
                decelerationRate="normal"
                showsHorizontalScrollIndicator={false}
                scrollEventThrottle={10}
                snapToInterval={boxWidth}
                contentOffset={{ x: -120, y: 0 }}
                ItemSeparatorComponent={() => <View style={{width: 20}} />}
                
                data={props.cards}
                renderItem={({item, index}) => {
                    return (
                        <View key={index}> 
                            <CatCard 
                                catImage={item.catImage}
                                catName={item.catName}
                                iconName1={item.iconName1}
                                field1={item.field1}
                                info1={item.info1}
                                iconName2={item.iconName2}
                                field2={item.field2}
                                info2={item.info2}
                            />
                        </View>
                        
                    );
                }}
        />

    );
};

export { Carousel }