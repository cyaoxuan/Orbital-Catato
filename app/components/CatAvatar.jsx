import { StyleSheet, TouchableOpacity, View } from "react-native";
import { Avatar } from "react-native-paper";
import { BodyText, TitleText } from "./Text";

// Cat Avatar with circular image and name
// props: photoURL (string), size (image size || number), variant (font size || string), name (string)
const CatAvatar = (props) => {
    return (
        <View style={styles.avatarContainer}>
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </View>
    );
};

// Cat Avatar with circular image and name, touchable
// props: photoURL (string), size (image size || number), variant (font size || string), name (string), onPress (callback)
const TouchableCatAvatar = (props) => {
    return (
        <TouchableOpacity
            testID="cat-avatar"
            style={styles.avatarContainer}
            activeOpacity={0.7}
            onPress={props.onPress}
        >
            <Avatar.Image
                testID="avatar-image"
                source={
                    props.photoURL
                        ? { uri: props.photoURL }
                        : require("../../assets/placeholder.png")
                }
                size={props.size || 100}
            />
            <BodyText variant={props.variant} text={props.name || "Name"} />
        </TouchableOpacity>
    );
};

export { CatAvatar, TouchableCatAvatar };

const styles = StyleSheet.create({
    avatarContainer: {
        alignItems: "center",
    },
});
