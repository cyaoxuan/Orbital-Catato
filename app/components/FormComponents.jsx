import { useState } from "react";
import { View, StyleSheet } from "react-native";
import {
    Button,
    HelperText,
    Menu,
    RadioButton,
    Text,
    TextInput,
} from "react-native-paper";
import { SelectList } from "react-native-dropdown-select-list";
import NumericInput from "react-native-numeric-input";
import RNDateTimePicker from "@react-native-community/datetimepicker";
import { timeOptions } from "../data/DateTimeOptions";

const titleVariant = "titleMedium";
const bodyVariant = "bodyMedium";

// For Location
const DropdownList = (props) => {
    const [selected, setSelected] = useState("");

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <SelectList
                setSelected={
                    props.setSelected
                        ? props.setSelected
                        : (val) => setSelected(val)
                }
                data={props.data || [{ key: "1", value: "value" }]}
                save="value"
            />
        </View>
    );
};

// Text inputs, description reasons, name, features etc
// Required field
const FormInput = (props) => {
    const [value, setValue] = useState("");

    return (
        <View style={styles.container}>
            <TextInput
                style={styles.formInput}
                disabled={props.disabled}
                multiline={props.multiline}
                label={props.label || "Label"}
                placeholder={props.placeholder}
                value={props.value ? props.value : value}
                onChangeText={
                    props.onChangeText ? props.onChangeText : setValue
                }
                error={props.value ? !props.value.trim() : !value.trim()}
            />
            <HelperText
                type="error"
                padding="none"
                visible={props.value ? !props.value.trim() : !value.trim()}
            >
                Error: {props.errorText || "Error"}
            </HelperText>
        </View>
    );
};

// For year
const NumberSpinner = (props) => {
    const [number, setNumber] = useState(new Date().getFullYear());

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <NumericInput
                type="up-down"
                rounded
                editable={false}
                minValue={props.min}
                maxValue={props.max}
                value={props.value ? props.value : number}
                onChange={
                    props.onChange
                        ? props.onChange
                        : (num) => {
                              setNumber(num);
                          }
                }
                totalWidth={120}
                totalHeight={40}
            />
        </View>
    );
};

// For time
const TimeInput = (props) => {
    const today = new Date();
    const [date, setDate] = useState(new Date());
    const [show, setShow] = useState(false);

    const onChange = (event, selectedDate) => {
        const currentDate = selectedDate;
        setShow(false);
        setDate(currentDate);
    };

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <Button
                mode="contained-tonal"
                onPress={props.onPress ? props.onPress : setShow}
                style={{ width: "50%", margin: 4 }}
            >
                Pick Time{" "}
            </Button>
            <Text variant={bodyVariant}>
                Selected Time:{" "}
                {props.displayTime
                    ? props.displayTime.toLocaleTimeString("en-GB", timeOptions)
                    : date.toLocaleTimeString("en-GB", timeOptions)}
            </Text>
            {(props.displayTime ? props.displayTime > today : date > today) && (
                <Text variant={bodyVariant} style={{ color: "crimson" }}>
                    Error: Selected future time! Are you a time traveller?
                </Text>
            )}
            {(props.show ? props.show : show) && (
                <RNDateTimePicker
                    mode="time"
                    display="spinner"
                    is24Hour={true}
                    value={props.value ? props.value : date}
                    onChange={props.onChange ? props.onChange : onChange}
                />
            )}
        </View>
    );
};

// For gender, sterilised, concerns, foster
const TwoRadioInput = (props) => {
    const [selected, setSelected] = useState("First");

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>{props.titleText}</Text>
            <RadioButton.Group
                value={props.value ? props.value : selected}
                onValueChange={
                    props.onValueChange
                        ? props.onValueChange
                        : (value) => setSelected(value)
                }
            >
                <View style={styles.radioButtonContainer}>
                    <Text variant={bodyVariant}>{props.firstText}</Text>
                    <RadioButton value={props.firstValue || "First"} />
                </View>
                <View style={styles.radioButtonContainer}>
                    <Text variant={bodyVariant}>{props.secondText}</Text>
                    <RadioButton value={props.secondValue || "Second"} />
                </View>
            </RadioButton.Group>
        </View>
    );
};

// To upload photos in form
const UploadPhotos = (props) => {
    const [visible, setVisible] = useState(false);
    const openMenu = () => setVisible(true);
    const closeMenu = () => setVisible(false);

    return (
        <View style={styles.container}>
            <Text variant={titleVariant}>Upload Photos:</Text>
            <View>
                <Menu
                    style={{ marginTop: -50 }}
                    visible={visible}
                    onDismiss={closeMenu}
                    anchor={
                        <Button
                            mode="contained-tonal"
                            icon="camera"
                            onPress={openMenu}
                            style={{ width: "50%", margin: 4 }}
                        >
                            Upload{" "}
                        </Button>
                    }
                >
                    <Menu.Item
                        leadingIcon="camera"
                        onPress={props.cameraOnPress}
                        title="Camera"
                    />
                    <Menu.Item
                        leadingIcon="image"
                        onPress={props.galleryOnPress}
                        title="Gallery"
                    />
                </Menu>
            </View>
            <Text variant={bodyVariant}>
                Image: {props.photoURI ? props.photoURI : ""}
            </Text>
        </View>
    );
};

export {
    DropdownList,
    FormInput,
    NumberSpinner,
    TimeInput,
    TwoRadioInput,
    UploadPhotos,
};

const styles = StyleSheet.create({
    formInput: {
        fontSize: 16,
        width: "100%",
        backgroundColor: "transparent",
        paddingHorizontal: 0,
        alignContent: "center",
    },

    container: {
        width: "90%",
        margin: 4,
    },

    radioButtonContainer: {
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
        paddingHorizontal: 4,
    },

    icon: {
        marginHorizontal: 8,
    },
});
