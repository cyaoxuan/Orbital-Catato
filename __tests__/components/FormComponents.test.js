import { FormInput } from "../../app/components/FormComponents";
import { cleanup, render } from "@testing-library/react-native";

afterEach(cleanup)

describe("<FormInput />", () => {
    it("renders successfully", () => {
        const formInput = render(<FormInput />);
        expect(formInput).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<FormInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("displays label prop", () => {
        const { queryAllByText } = render(<FormInput label="Meow!" />);
        expect(queryAllByText("Meow!")).not.toBeNull();
    });
});