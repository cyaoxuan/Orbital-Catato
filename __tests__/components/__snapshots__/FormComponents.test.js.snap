// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownList /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  />
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": undefined,
            },
            undefined,
          ]
        }
      >
        Select option
      </Text>
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<FormInput /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "alignContent": "center",
          "backgroundColor": "transparent",
          "width": "100%",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(179, 38, 30, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 2,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 56,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            {
              "color": "rgba(179, 38, 30, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "opacity": 0,
              "position": "absolute",
              "textAlign": "left",
              "top": 30,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
          testID="text-input-flat-label-active"
        >
          Label
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            {
              "color": "rgba(179, 38, 30, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "opacity": 1,
              "position": "absolute",
              "textAlign": "left",
              "top": 30,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
          testID="text-input-flat-label-inactive"
        >
          Label
        </Text>
      </View>
      <TextInput
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(179, 38, 30, 1)"
        style={
          [
            {
              "margin": 0,
            },
            {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            {
              "height": 56,
            },
            {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 1,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Error: 
    Error
  </Text>
</View>
`;

exports[`<NumberSpinner /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "grey",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        {
          "borderColor": "#d4d4d4",
          "height": 40,
          "width": 120,
        },
        {
          "borderRadius": 7.199999999999999,
        },
        {},
      ]
    }
  >
    <TextInput
      editable={false}
      keyboardType="numeric"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      returnKeyType="done"
      style={
        [
          {
            "padding": 0,
            "textAlign": "center",
          },
          {
            "borderRightColor": "#d4d4d4",
            "borderRightWidth": 2,
            "color": "black",
            "fontSize": 15.2,
            "height": 40,
            "width": 72,
          },
          {},
        ]
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="2023"
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRightColor": "#d4d4d4",
            "borderRightWidth": 1,
            "width": 48,
          },
          {
            "borderBottomRightRadius": 7.199999999999999,
            "borderTopRightRadius": 7.199999999999999,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 15.2,
              },
              [
                {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "fontWeight": "900",
                },
                {},
                {},
                {},
              ],
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 15.2,
              },
              [
                {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "fontWeight": "900",
                },
                {},
                {},
                {},
              ],
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<TimeInput /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(232, 222, 248, 1)",
        "borderRadius": 20,
        "margin": 4,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "50%",
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(29, 25, 43, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
            testID="button-text"
          >
            Pick Time
             
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Selected Time:
     
    07:00
     
    (SGT, GMT+8)
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    *Time Picker uses device local time, which may cause discrepancies with selected time
  </Text>
</View>
`;

exports[`<TwoRadioInput /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  />
  <View
    accessibilityRole="radiogroup"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      />
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "borderColor": "rgba(103, 80, 164, 1)",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(103, 80, 164, 1)",
                  "borderRadius": 5,
                  "height": 10,
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      />
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "borderColor": "rgba(73, 69, 79, 1)",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<UploadPhotos /> renders correctly 1`] = `
<View
  style={
    {
      "margin": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Upload Photos:
  </Text>
  <View>
    <View
      collapsable={false}
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(232, 222, 248, 1)",
            "borderRadius": 20,
            "margin": 4,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": "50%",
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(232, 222, 248, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "marginLeft": 12,
                      "marginRight": -4,
                    },
                    {
                      "marginLeft": 16,
                      "marginRight": -16,
                    },
                    false,
                  ]
                }
                testID="button-icon-container"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(29, 25, 43, 1)",
                        "fontSize": 18,
                      },
                      [
                        {
                          "lineHeight": 18,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰄀
                </Text>
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(29, 25, 43, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="button-text"
              >
                Upload
                 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Image: 
  </Text>
</View>
`;
